### PR TITLE
[VPU] Rename INTEL_VPU to INTEL_MYRIAD, move thirdparty and vpu_dependencies

### PR DESCRIPTION
**Pull request by AntonDudchenko**
### Details:
 - *Rename INTEL_VPU to INTEL_MYRIAD due to possible problems with naming the VPUX plugin*
 - *Move thirdparty components into plugin folder*

### Tickets:
 - *71049*

### CODE REVIEW

File: cmake/dependencies.cmake [hunk 1]
```diff
@@ -51,8 +51,8 @@ if(CMAKE_CROSSCOMPILING AND CMAKE_HOST_SYSTEM_NAME MATCHES Linux AND CMAKE_HOST_
     update_deps_cache(SYSTEM_PROTOC "${SYSTEM_PROTOC}" "Path to host protoc for ONNX Importer")
 endif()
 
-if(ENABLE_INTEL_VPU)
-    include(${IE_MAIN_SOURCE_DIR}/cmake/vpu_dependencies.cmake)
+if(ENABLE_INTEL_MYRIAD)
+    include(${OpenVINO_SOURCE_DIR}/src/plugins/intel_myriad/vpu_dependencies.cmake)
```
Comment (ilya-lavrenov) [inline cmake/dependencies.cmake:55]: `vpu_dependencies` => `myriad_dependencies` ?

File: inference-engine/thirdparty/CMakeLists.txt [hunk 2]
```diff
@@ -2,10 +2,6 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-if(ENABLE_INTEL_VPU)
```
Comment (ilyachur) [inline inference-engine/thirdparty/CMakeLists.txt:5]: Please remove this cmake script

**Comment by AntonDudchenko** [comment]
@DariaMityagina @KorA6 Could you please take a look,
@Maxim-Doronin FYI

File: inference-engine/thirdparty/CMakeLists.txt (hunk 2 above)
Comment (AntonDudchenko) [reply inference-engine/thirdparty/CMakeLists.txt:5]: Done

File: CODEOWNERS [hunk 3]
```diff
@@ -49,7 +49,7 @@ Jenkinsfile  @openvinotoolkit/openvino-admins
 /src/plugins/intel_gpu/  @openvinotoolkit/openvino-ie-gpu-maintainers @openvinotoolkit/openvino-ie-gpu-developers
 
 # IE VPU:
-/src/plugins/intel_vpu  @openvinotoolkit/openvino-ie-vpu-maintainers
+/src/plugins/intel_myriad  @openvinotoolkit/openvino-ie-vpu-maintainers
 /src/inference/include/ie/vpu/  @openvinotoolkit/openvino-ie-vpu-maintainers
 /inference-engine/thirdparty/movidius/  @openvinotoolkit/openvino-ie-vpu-maintainers
```
Comment (ilyachur) [inline CODEOWNERS:54]: I think it also should be changed.

**Review by KorA6 — APPROVED** [review]

**Review by DariaMityagina — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/814

**Pull request merged**